File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int totalLength = temperatures.length;
        int[] res = new int[totalLength];
        Stack<int[]> stack = new Stack<>(); // [temp, index]

        for (int i = 0; i < totalLength; i++) {
            int t = temperatures[i];
            while (!stack.isEmpty() && t > stack.peek()[0]) {
                int[] pair = stack.pop();
                res[pair[1]] = i - pair[1];
            }
            stack.push(new int[]{t, i});
        }

        return res;
        
    }
}

File: Data Structures & Algorithms/generate-parentheses/submission-1.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        res = []
        stack = []
        def dfs(opened, closed, stack):
            if closed == opened == n:
                res.append("".join(stack))
                return
            
            # add (
            if opened < n:
                stack.append("(")
                dfs(opened + 1, closed, stack)
                stack.pop()

            # add )
            if closed < opened:
                stack.append(")")
                dfs(opened, closed + 1, stack)
                stack.pop()

        dfs(0,0,stack)
        return res